tests/conftest.py

```diff
@@ -19,6 +26,41 @@ def spark_session() -> SparkSession:
     return SparkSession.getActiveSession() or get_local_session()
 
 
+@pytest.fixture(scope="session")
+def checkpoint_directory():
+    """Fixture to validate and remove the checkpoint directory after tests.
+
+    To avoid test failures due to non-unique directories, the user should add a
+    subdirectory to this path when using this fixture.
+
+    Example:
+        >>> dir = checkpoint_directory + subdir
+    """
+    checkpoint_dir = "/tmp/checkpoint/"
+
+    yield checkpoint_dir
+
+    # Remove the checkpoint directory if it exists
+    if os.path.exists(checkpoint_dir):
+        shutil.rmtree(checkpoint_dir)
+        print(f"Checkpoint directory {checkpoint_dir} has been removed.")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def cleanup_logging_handlers():
+    """Fixture to cleanup logging handlers after tests.
+
+    Prevents logging errors at the end of the report.
```

Comment:
Is it causing any harm? If not I'd suggest to not implement it like this as it's just silencing any possible error in handlers, and with future updates might introduce unwanted behavior.

Reply:
> Is it causing any harm? 
To my eyes, yes. It's also related to pytest logs being cluttered.

This is a long standing known issue of pytest and a sufficiently deep rabbit hole: https://github.com/pytest-dev/pytest/issues/5502#issuecomment-1803676152.

There are some situations/tests (pardon the imprecision; I delved into this too many years ago) where pytest attempts to write to closed streams AFTER the session is over.

This results in error messages being printed at the very end of the pytest recap, which provide no useful information and clutter the output.

This fixture prevents this from happening, since it's scope is `session` it shouldn't tamper with anything test related.